python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -12,266 +12,289 @@
     )
 
 
+# A metadata key used to mark a dataclass field as a metric.
+_IS_FIELD_METRIC_KEY = "__is_metric"
+# Metadata keys used to store information about a metric.
+_METRIC_FIELD_DESCRIPTION_KEY = "__metric_description"
+_METRIC_FIELD_METRICS_GROUP_KEY = "__metric_metrics_group"
+_METRIC_FIELD_IS_MAP_ONLY_KEY = "__metric_is_map_only"
+
+_METRICS: List["Metric"] = []
+
+
+@dataclass(frozen=True)
+class Metric:
+    """Metadata for a metric.
+
+    Args:
+        name: The name of the metric.
+        description: A human-readable description of the metric, also used as the chart
+            description on the Ray Data dashboard.
+        metrics_group: The group of the metric, used to organize metrics into groups in
+            'StatsActor' and on the Ray Data dashboard.
+        map_only: Whether the metric is only measured for 'MapOperators'.
+    """
+
+    name: str
+    description: str
+    metrics_group: str
+    map_only: bool = False
+
+
+def metricfield(
```

Comment:
nit, `metric_filed` for better readability? same for `metric_property`

Reply:
Yeah, I'm down to make that change.

Didn't use underscores for consistency with under names like `classmethod`.

Reply:
Fixed.